### PR TITLE
feat(portfolio) : add download api endpoint

### PR DESCRIPTION
**Pull request by beesk135**
- add a validator to model property
- add auth e2e test
- add deleteUser function in admin

### CODE REVIEW

File: src/admin/admin.controller.ts [hunk 1]
```diff
@@ -69,4 +71,17 @@ export class AdminController {
...
         return this.userService.find({ role: UserRole.Tutor });
     }
+
+    @Get('portfolio/list')
+    async listPortfolio() {
+        const listFiles : String[] = await this.fileService.listFiles("");
+        return listFiles.map((name : string)=>{
+            return name.split("/")[1]
+        })
+    }
```
Comment (UtopiaBeam) [inline src/admin/admin.controller.ts:81]: - `String` should be `string`.

File: src/file/file.service.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ export class FileService {
         return { contentType, signedUrl };
     }
 
-    async listFiles(prefix: string): Promise<string[]> {
+    async listFiles(prefix?: string): Promise<string[]> {
         const [files] = await this.bucket.getFiles({ prefix });
         return files.map(f => f.name);
     }
```
Comment (UtopiaBeam) [inline src/file/file.service.ts:32]: `prefix` should not be optional.

File: package.json [hunk 3]
```diff
@@ -16,7 +16,7 @@
     "test": "jest",
     "test:watch": "jest --watch",
     "test:cov": "jest --coverage",
-    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
+    "test:debug": "node --inspect-brk -r tsconfig-paths/auth/register -r ts-node/auth/register node_modules/.bin/jest --runInBand",
```
Comment (UtopiaBeam) [inline package.json:19]: Why change script here?

File: test/auth.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+            status: UserStatus.Active,
+            credit: 0,
+            verified: false,
+        };
+        return await request(app)
+            .post('/auth/register')
+            .send(user)
+            .expect(({ body }) => {
+                // console.log(body)
+            })
```
Comment (UtopiaBeam) [inline test/auth.e2e-spec.ts:53]: What does this mean?

File: package.json (hunk 3 above)
Comment (beesk135) [reply package.json:19]: wow idk what's that? I will it for ya

File: src/model/user.model.ts [hunk 5]
```diff
@@ -42,6 +47,8 @@ export class User {
 
     @ApiProperty({ required: true })
     @IsString()
+    @IsNotEmpty()
+    @IsByteLength(8)
```
[the remark this answers is not in the document]
Comment (beesk135) [reply src/model/user.model.ts:51]: ok thanks

File: src/admin/admin.controller.ts [hunk 6]
```diff
@@ -69,4 +72,22 @@ export class AdminController {
...
+    @Get('portfolio/download/:id')
+    downloadPortfolio(@Param('id') id: string) {
+        return this.userService.downloadPortfolio(`portfolio/${id}`);
+    }
+
+    @Delete('deleteUser')
+    deleteUser(@Body() userDTO: Partial<User>) {
+        return this.userService.deleteUser(userDTO);
+    }
 }
```
[the remark this answers is not in the document]
Comment (beesk135) [reply src/admin/admin.controller.ts:93]: ok

File: test/auth.e2e-spec.ts [hunk 7]
```diff
@@ -0,0 +1,279 @@
+import * as request from 'supertest';
+import { Test, TestingModule } from '@nestjs/testing';
+// import { AuthModule } from '../auth/auth.module';
```
[the remark this answers is not in the document]
Comment (beesk135) [reply test/auth.e2e-spec.ts:3]: ok

File: test/auth.e2e-spec.ts (hunk 4 above)
Comment (beesk135) [reply test/auth.e2e-spec.ts:53]: forget to remove it

Comment (beesk135) [reply test/auth.e2e-spec.ts:53]: don't care this e2e, it's not the last version

File: src/user/user.service.ts [hunk 8]
```diff
@@ -90,4 +81,12 @@ export class UserService {
     uploadPortfolio(name: string, file: FileDTO) {
         return this.fileService.upload(name, file);
     }
+
+    downloadPortfolio(name: string) {
+        return this.fileService.getFile(name);
+    }
+
+    deleteUser(id: string) {
+        return this.model.findOneAndDelete({ _id: id }).exec();
```
Comment (theoatboy139) [inline src/user/user.service.ts:90]: I think this one should use `findByIdAndDelete`

**Pull request merged**